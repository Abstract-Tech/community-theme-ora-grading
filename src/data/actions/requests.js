import { StrictDict } from 'utils';
import { createActionFactory } from './utils';

export const dataKey = 'requests';
const createAction = createActionFactory(dataKey);

export const startRequest = createAction('startRequest');
export const completeRequest = createAction('completeRequest');
export const failRequest = createAction('failRequest');

export default StrictDict({
  startRequest,
  completeRequest,
  failRequest,
});
