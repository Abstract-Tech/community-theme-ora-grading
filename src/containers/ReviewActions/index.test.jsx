import React from 'react';
import { shallow } from 'enzyme';

import actions from 'data/actions';
import selectors from 'data/selectors';

import { ReviewActions, mapStateToProps, mapDispatchToProps } from '.';

jest.mock('@edx/paragon', () => ({
  ActionRow: () => 'ActionRow',
  Button: () => 'Button',
}));
jest.mock('data/selectors', () => ({
  __esModule: true,
  default: {
    app: { showRubric: (state) => ({ showRubric: state }) },
    grading: {
      selected: {
        username: (state) => ({ username: state }),
        gradingStatus: (state) => ({ gradingStatus: state }),
      },
    },
  },
}));
jest.mock('components/StatusBadge', () => 'StatusBadge');
jest.mock('./components/StartGradingButton', () => 'StartGradingButton');
jest.mock('./components/SubmissionNavigation', () => 'SubmissionNavigation');

describe('ReviewActions component', () => {
  describe('component', () => {
    const props = {
      gradingStatus: 'grading-status',
      username: 'test-username',
      showRubric: false,
    };
    beforeEach(() => {
      props.toggleShowRubric = jest.fn().mockName('this.props.toggleShowRubric');
    });
    test('snapshot: do not show rubric', () => {
      expect(shallow(<ReviewActions {...props} />)).toMatchSnapshot();
    });
    test('snapshot: show rubric', () => {
      expect(shallow(<ReviewActions {...props} showRubric />)).toMatchSnapshot();
    });
  });
  describe('mapStateToProps', () => {
    let mapped;
    const testState = { some: 'test-state' };
    beforeEach(() => {
      mapped = mapStateToProps(testState);
    });
    test('username loads from grading.selected.username', () => {
      expect(mapped.username).toEqual(selectors.grading.selected.username(testState));
    });
    test('gradingStatus loads from grading.selected.gradingStatus', () => {
      expect(mapped.gradingStatus).toEqual(selectors.grading.selected.gradingStatus(testState));
    });
    test('showRubric loads from app.showRubric', () => {
      expect(mapped.showRubric).toEqual(selectors.app.showRubric(testState));
    });
  });
  describe('mapDispatchToProps', () => {
    it('loads toggleShowRubric from actions.app.toggleShowRubric', () => {
      expect(mapDispatchToProps.toggleShowRubric).toEqual(actions.app.toggleShowRubric);
    });
  });
});
