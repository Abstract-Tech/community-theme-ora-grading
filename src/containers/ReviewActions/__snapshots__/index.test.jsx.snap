// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewActions component component snapshot: do not show rubric 1`] = `
<div>
  <ActionRow
    className="review-actions"
  >
    <span
      className="review-actions-username"
    >
      <span
        className="lead"
      >
        test-username
      </span>
      <StatusBadge
        className="review-actions-status mr-3"
        status="grading-status"
      />
      <span
        className="small"
      >
        Score: 3/10
      </span>
    </span>
    <div
      className="review-actions-group"
    >
      <Button
        onClick={[MockFunction this.props.toggleShowRubric]}
        variant="outline-primary"
      >
        Show
         Rubric
      </Button>
      <StartGradingButton />
      <SubmissionNavigation />
    </div>
  </ActionRow>
</div>
`;

exports[`ReviewActions component component snapshot: show rubric, no score 1`] = `
<div>
  <ActionRow
    className="review-actions"
  >
    <span
      className="review-actions-username"
    >
      <span
        className="lead"
      >
        test-username
      </span>
      <StatusBadge
        className="review-actions-status mr-3"
        status="grading-status"
      />
      <span
        className="small"
      />
    </span>
    <div
      className="review-actions-group"
    >
      <Button
        onClick={[MockFunction this.props.toggleShowRubric]}
        variant="outline-primary"
      >
        Hide
         Rubric
      </Button>
      <StartGradingButton />
      <SubmissionNavigation />
    </div>
  </ActionRow>
</div>
`;
